index.bs

```diff
@@ -403,107 +410,100 @@ these steps:
 
 </div>
 
-## The {{FileSystemWriter}} interface ## {#api-filesystemwriter}
+## The {{FileSystemWritableFileStream}} interface ## {#api-filesystemwritablefilestream}
 
 <xmp class=idl>
 [Exposed=(Window,Worker), SecureContext]
-interface FileSystemWriter {
+interface FileSystemWritableFileStream : WritableStream {
   Promise<void> write(unsigned long long position, (BufferSource or Blob or USVString) data);
```

Comment:
You shouldn't define write() and close(); users should call those via the writer.

To customize the behavior of the writer, instead you need to define (in the definition of createWritable) what the underlying sink algorithms are. See https://fetch.spec.whatwg.org/#concept-construct-readablestream and its call sites for some rough guidance (although that's for readable; for writable you'd be using https://streams.spec.whatwg.org/#create-writable-stream).

Reply:
What do you think of defining `write()` and `truncate()`, so that `FileSystemWritableFileStream` can be used directly?

For instance, one could do:

```js
const stream = await handle.createWritable();
await stream.write(new Blob(['foo']); // New method, peer to `seek` and `truncate`.
await stream.close(); // Overrided method from WritableStream.
```

At the same time, the underlying sink algorithms will be implemented to accept `WriteParams` to execute commands, or to take byte-like data.